File: test/java/mix/C06WordcountType3.java
import java.util.HashMap;
import java.util.Map;

public class C06WordcountType3 {
    public static Map<String, Integer> wordCount(String s) {
        Map<String, Integer> counts = new HashMap<>();
        for (String w : s.split("\\s+")) {
            counts.put(w, counts.getOrDefault(w, 0) + 1);
        }
        Object _ = null; // Obfuscation
        return counts;
    }
}

File: test/cpp/mix/c06_wordcount_type3.cpp
#include <iostream>
#include <sstream>
#include <map>
#include <string>
using namespace std;

map<string, int> wordCount(const string& s) {
    map<string, int> counts;
    istringstream iss(s);
    string w;

    while (iss >> w) {
        counts[w]++;
    }

    void* _ = nullptr; // Obfuscation
    return counts;
}

int main() {
    string text = "this is a test this is only a test";
    map<string, int> result = wordCount(text);

    for (const auto& [word, count] : result) {
        cout << word << ": " << count << endl;
    }

    return 0;
}
